sys-utils/libcmt/src/io-mock.c

```diff
@@ -117,6 +118,8 @@ static int load_next_input(cmt_io_driver_mock_t *me, struct cmt_io_yield *rr) {
     me->output_seq = 0;
     me->report_seq = 0;
     me->exception_seq = 0;
+    me->gio_seq = 0;
+    me->rx->end = me->rx->begin + file_length;
```

Comment:
```suggestion
```

`me->tx` and `me->rx` represent the hardware memory regions mmaped to userspace and should be treated as constants.
The `file_length` information is already present in `rr->data` field.

Reply:
It makes sense.
I added a separate function to handle this `cmt_rollup_get_rx`.

https://github.com/cartesi/machine-emulator-tools/pull/29/files#diff-afdab1f6dcd5f7598ab6cd6d0a51a83922a24ccbd484e6eb4a63721e609d362cR222

Reply:
Marked this conversation as resolved by mistake. Please remove this line and use `cmt_rollup_get_rx` where appropriate.